src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx

```diff
@@ -170,9 +164,20 @@ export class DashboardAppController {
 
     // The hash check is so we only update the time filter on dashboard open, not during
     // normal cross app navigation.
-    if (dashboardStateManager.getIsTimeSavedWithDashboard() && !hasInheritedGlobalStateFromUrl) {
+    const initialGlobalStateInUrl = kbnUrlStateStorage.get<QueryState>('_g');
+    const hasInheritedTimeRange = Boolean(initialGlobalStateInUrl?.time);
+    if (dashboardStateManager.getIsTimeSavedWithDashboard() && !hasInheritedTimeRange) {
       dashboardStateManager.syncTimefilterWithDashboard(timefilter);
     }
+    // starts syncing `_g` portion of url with query services
+    // note: dashboard_state_manager.ts syncs `_a` portion of url
+    // it is important to start this syncing after `dashboardStateManager.syncTimefilterWithDashboard(timefilter);` above is run,
+    // otherwise it will case redundant browser history record
+    const { stop: stopSyncingQueryServiceStateWithUrl } = syncQueryStateWithUrl(
```

Comment:
This was moved AFTER setup of `dashboardStateManager` and `dashboardStateManager.syncTimefilterWithDashboard(timefilter);` because otherwise it is causing redundant browser history record. 

The side effect of this change is that now in dashboard order of state query params in url could be different: 
e.g.: 

before:
`/dashboard/id/?_g=".."&_a=""`
but with this change:
`/dashboard/id/?_a=".."&_g=""`

I don't think it could be considered a breaking change as order of query params should not matter. 
This affected one functional test where url parsing relied on order of those params: 
https://github.com/elastic/kibana/pull/60087/files#diff-c887a49735ad5390c82f29319cc187ebR48

Reply:
Extracted to separate pr in scope of: https://github.com/elastic/kibana/issues/61230